### ISSUE TITLE
File->Open does not update window content correctly - Appeared first in v7-3-638

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vim/vim#91]
```

Problem is not present in tag v7-3-637. Problem appeared first in tag v7-3-638.

What steps will reproduce the problem?
Using gvim:
1. Go to v7-3-638 vim source directory
2. File->Open and select Filelist
3. Content shows correctly
4. File->Open and select README.txt
5. Contents are not updated, section of Filelist still shows
6. Move mouse wheel or cursor, contents are now updated correctly

What is the expected output? What do you see instead?
Exepcted output is content of newly opened file. I see part of old file until 
screen refreshes by moving cursor.

What version of the product are you using? On what operating system?
Linux Mint 13 with MATE 1.2 or 1.4
gvim compiled from source with: ./configure --enable-gui=gtk2 
--enable-rubyinterp=yes --with-features=huge
Kernel:  3.2.0-32-generic #51-Ubuntu SMP

Please provide any additional information below.

Problem is not present in tag v7-3-637. Problem appeared first in tag v7-3-638.
```

Original issue reported on code.google.com by `yerv...@gmail.com` on 3 Nov 2012 at 1:33

**Comment by fabd** [comment]
If this is any help, on my Ubuntu 13.04 development machine, I had to regress the Virtual Box Guest Additions to 4.2.12 to get rid of the problem. Absolutely critical bug that made it impossible to use gvim otherwise. Very, very confusing when you switch between many tabs of code and part of the code is not redrawn, so you are looking at content from two different buffers..

So in regard to the comment above, it may be that the "two other Ubuntu 14.04 installations" were using an older Guest Additions.

**Comment by wsdjeg** [comment]
is there any differeces between vim and gvim in ubuntu system,i have navel used gvim.

**Comment by fabd** [comment]
More info:

I've just compiled v7.3.637 (git branch from the older commit). Version is confirmed on the About page. 

The patch v7.3.638 changes a couple lines or so in src/screen.c right ? Which OP here assumes is what causes the problem.

gvim still presents the issue in Ubuntu 14.04.03 LTS, guest under a Windows 7 host, Virtual Box 5.0.4 + Guest Additions.

I built like so:

    sudo apt-get build-dep vim-gnome
    ./configure --with-gui=gtk2 # or gnome
    make -j 4
    sudo make install  

Simply open two text files that fill the screen (ie. 150+ lines), then rapidly ctrl+home/end or ctrl+pgup/dn to move between top/bottom, or switch tabs. Either way within seconds a half redraw will appear. Sometimes it's vertically split, sometimes it draws only a few columns on the left side.

From what I gather this bug tend to be reported by people running a virtual machine with Windows 7 as host, and Ubuntu as the guest.

**Comment by thomasgubler** [comment]
Same problem on Ubuntu 14.04. My workaround for now is: https://bitbucket.org/equalsraf/vim-qt/wiki/Home

**Comment by benbaez** [comment]
I have the same issue as GoogleCodeExporter using VMware Workstation 12.5 and Ubuntu 16.04 patched to latest.  Existed in Ubuntu 14 and Workstation 12 as well.  Host is a HP Z620.  This only occurs on this host all 3 of Ubuntu I am running.  Haven't noticed the issue on CentOS.  Using --sync option is a workaround.

**Comment by nuko8** [comment]
Roughly speaking, X clients cannot update their window content until a certain event from the X11 server arrives at their event queue and then drives the event loop to invoke the callback function for update.

As for the GTK2 GUI, the callback function for update is quite passive. It is associated with update events at initialization and never dissociated from them until the process terminates.  In addition, there’s no code at all in the implementation to block update events at runtime.

Moreover, as indicated in the top post:

> 1. Move mouse wheel or cursor, contents are now updated correctly

and in @chrisbra's [gtk2_force_redraw.diff](https://storage.googleapis.com/google-code-attachments/vim/issue-91/comment-4/gtk2_force_redraw.diff), the callback function works as expected once it gets an update event or another event which induces update.

In short, just like any other X clients, the GUI is made not to exercise its veto over any update requests.

Accordingly, it is highly likely that either of the windowing system under a certain environment or the GTK library (or some interaction between them) prevents update events from being delivered to the GUI for the case in question.

If this is the case, there’s nothing left for the GUI to do with.

**Comment by chrisbra** [comment]
i am closing this, as this is no gvim issue.
